### PR TITLE
how-to/upgrade: update `from previous version` doc

### PR DESCRIPTION
**Pull request by juliezhang1112**
### What is changed, added or deleted? 
- Add a note on configuring the `tikv_status_port` parameter
- Update supported versions of TiDB 2.1 upgrading to TiDB 3.0
- Remove outdated information: TiDB 3.0 is not compatible with TiDB Binlog of Kafka version

### What is the related PR or file link(s)? 
https://github.com/pingcap/docs-cn/pull/1893
https://github.com/pingcap/docs-cn/pull/1896
https://github.com/pingcap/docs-cn/pull/2034
https://github.com/pingcap/docs-cn/pull/2048

### Which version does your change affect? 
dev, v3.0, v2.1, v3.1

### CODE REVIEW

**Comment by juliezhang1112** [comment]
PTAL @WalterWj Thanks.

**Comment by WalterWj** [comment]
LGTM

**Comment by WalterWj** [comment]
> PTAL @WalterWj Thanks.

LGTM

File: dev/how-to/upgrade/from-previous-version.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ category: how-to
 
 # TiDB 3.0 Upgrade Guide
 
-This document is targeted for users who want to upgrade from TiDB 2.0 (2.0.1 or later versions) or TiDB 2.1 RC to TiDB 3.0. TiDB 3.0 is compatible with [TiDB Binlog of Kafka Version](/dev/reference/tidb-binlog/tidb-binlog-kafka.md) and [TiDB Binlog of Cluster Version](/dev/reference/tidb-binlog/overview.md).
+This document is targeted for users who want to upgrade from TiDB 2.0 (2.0.1 or later versions) or TiDB 2.1 (2.1 RC1 or later versions) to TiDB 3.0 or the master version. Currently, TiDB 3.0 and the master version are compatible with [TiDB Binlog of Cluster Version](/dev/reference/tidb-binlog/overview.md).
```
Comment (lilin90) [inline dev/how-to/upgrade/from-previous-version.md]: How about only keeping the description for master in `dev`?

```suggestion
This document is targeted for users who want to upgrade from TiDB 2.0 (2.0.1 or later versions) or TiDB 2.1 (2.1 RC1 or later versions) to the master version. Currently, TiDB master version is compatible with [TiDB Binlog of Cluster Version](/dev/reference/tidb-binlog/overview.md).
```

File: dev/how-to/upgrade/from-previous-version.md [hunk 2]
```diff
@@ -160,6 +160,22 @@ If you have previously customized the configuration file of TiDB cluster compone
...
+        TiKV1-2 ansible_host=172.16.10.4 deploy_dir=/data2/deploy tikv_port=20172 tikv_status_port=20182 labels="host=tikv1"
+        TiKV2-1 ansible_host=172.16.10.5 deploy_dir=/data1/deploy tikv_port=20171 tikv_status_port=20181 labels="host=tikv2"
+        TiKV2-2 ansible_host=172.16.10.5 deploy_dir=/data2/deploy tikv_port=20172 tikv_status_port=20182 labels="host=tikv2"
+        TiKV3-1 ansible_host=172.16.10.6 deploy_dir=/data1/deploy tikv_port=20171 tikv_status_port=20181 labels="host=tikv3"
+        TiKV3-2 ansible_host=172.16.10.6 deploy_dir=/data2/deploy tikv_port=20172 tikv_status_port=20182 labels="host=tikv3"
+    ```
+
+    > **Note:**
+    >
+    > If you upgrade to TiDB 3.0 with a cluster topology of multiple TiKV instances (processes) on a single machine, configure the `tikv_status_port` parameter. Before configuring the parameter, check for port conflicts.
```
Comment (lilin90) [inline dev/how-to/upgrade/from-previous-version.md]:
```suggestion
    > If you upgrade to TiDB master with a cluster topology of multiple TiKV instances (processes) on a single machine, configure the `tikv_status_port` parameter. Before configuring the parameter, check for port conflicts.
```

File: dev/how-to/upgrade/from-previous-version.md [hunk 3]
```diff
@@ -160,6 +160,22 @@ If you have previously customized the configuration file of TiDB cluster compone
...
+
 ## Step 4: Download TiDB 3.0 binary to the Control Machine
 
 Make sure that `tidb_version = v3.0.0` in the `tidb-ansible/inventory.ini` file, and then run the following command to download TiDB 3.0 binary to the Control Machine:
```
Comment (lilin90) [inline dev/how-to/upgrade/from-previous-version.md]: I think the `tidb_version = v3.0.0` should be updated.

File: v3.0/how-to/upgrade/from-previous-version.md [hunk 4]
```diff
@@ -7,7 +7,7 @@ aliases: ['/docs/v3.0/how-to/upgrade/to-tidb-3.0/','/docs/v2.1/how-to/upgrade/to
 
 # TiDB 3.0 Upgrade Guide
 
-This document is targeted for users who want to upgrade from TiDB 2.0 (2.0.1 or later versions) or TiDB 2.1 RC to TiDB 3.0. TiDB 3.0 is compatible with [TiDB Binlog of Kafka Version](/v3.0/reference/tidb-binlog/tidb-binlog-kafka.md) and [TiDB Binlog of Cluster Version](/v3.0/reference/tidb-binlog/overview.md).
+This document is targeted for users who want to upgrade from TiDB 2.0 (2.0.1 or later versions) or TiDB 2.1 (2.1 RC1 or later versions) to TiDB 3.0 or the master version. Currently, TiDB 3.0 and the master version are compatible with [TiDB Binlog of Cluster Version](/v3.0/reference/tidb-binlog/overview.md).
```
Comment (lilin90) [inline v3.0/how-to/upgrade/from-previous-version.md]:
```suggestion
This document is targeted for users who want to upgrade from TiDB 2.0 (2.0.1 or later versions) or TiDB 2.1 (2.1 RC1 or later versions) to TiDB 3.0 (3.0.0 or later versions). Currently, TiDB 3.0 versions are compatible with [TiDB Binlog of Cluster Version](/v3.0/reference/tidb-binlog/overview.md).
```

File: v3.0/how-to/upgrade/from-previous-version.md [hunk 5]
```diff
@@ -161,6 +161,22 @@ If you have previously customized the configuration file of TiDB cluster compone
...
+        TiKV3-2 ansible_host=172.16.10.6 deploy_dir=/data2/deploy tikv_port=20172 tikv_status_port=20182 labels="host=tikv3"
+    ```
+
+    > **Note:**
+    >
+    > If you upgrade to TiDB 3.0 with a cluster topology of multiple TiKV instances (processes) on a single machine, configure the `tikv_status_port` parameter. Before configuring the parameter, check for port conflicts.
+
 ## Step 4: Download TiDB 3.0 binary to the Control Machine
 
 Make sure that `tidb_version = v3.0.0` in the `tidb-ansible/inventory.ini` file, and then run the following command to download TiDB 3.0 binary to the Control Machine:
```
Comment (lilin90) [inline v3.0/how-to/upgrade/from-previous-version.md]: Perhaps `tidb_version = v3.0.0` -> `tidb_version = v3.0.x`? @liubo0127

File: v3.1/how-to/upgrade/from-previous-version.md [hunk 6]
```diff
@@ -6,7 +6,7 @@ category: how-to
 
 # TiDB 3.0 Upgrade Guide
 
-This document is targeted for users who want to upgrade from TiDB 2.0 (2.0.1 or later versions) or TiDB 2.1 RC to TiDB 3.0. TiDB 3.0 is compatible with [TiDB Binlog of Kafka Version](/v3.1/reference/tidb-binlog/tidb-binlog-kafka.md) and [TiDB Binlog of Cluster Version](/v3.1/reference/tidb-binlog/overview.md).
+This document is targeted for users who want to upgrade from TiDB 2.0 (2.0.1 or later versions) or TiDB 2.1 (2.1 RC1 or later versions) to TiDB 3.0 or the master version. Currently, TiDB 3.0 and the master version are compatible with [TiDB Binlog of Cluster Version](/v3.1/reference/tidb-binlog/overview.md).
```
Comment (lilin90) [inline v3.1/how-to/upgrade/from-previous-version.md]: @liubo0127 @ericsyh Please take a look at the v3.1 upgrade guide. If only minor updates are needed, we can update it in this PR. If not, merge this first and then update it in another PR.

File: v3.1/how-to/upgrade/from-previous-version.md [hunk 7]
```diff
@@ -6,7 +6,7 @@ category: how-to
 
 # TiDB 3.0 Upgrade Guide
```
Comment (lilin90) [inline v3.1/how-to/upgrade/from-previous-version.md]:
```suggestion
# TiDB 3.1 Upgrade Guide
```

File: dev/how-to/upgrade/from-previous-version.md (hunk 3 above)
Comment (liubo0127) [reply dev/how-to/upgrade/from-previous-version.md]: agree.

File: v3.0/how-to/upgrade/from-previous-version.md (hunk 5 above)
Comment (liubo0127) [reply v3.0/how-to/upgrade/from-previous-version.md]: LGTM

File: dev/how-to/upgrade/from-previous-version.md (hunk 3 above)
Comment (liubo0127) [inline dev/how-to/upgrade/from-previous-version.md]:
```suggestion
Make sure that `tidb_version = latest` in the `tidb-ansible/inventory.ini` file, and then run the following command to download TiDB 3.0 binary to the Control Machine:
```

**Comment by lilin90** [comment]
@juliezhang1112 Please address comments.

**Comment by lilin90** [comment]
This issue is closed because changes in this PR have been covered in #1689.

**Pull request closed without merging**